pgml-extension/src/orm/snapshot.rs

```diff
@@ -256,22 +260,61 @@ impl Snapshot {
             .unwrap() as usize
     }
 
+    fn fully_qualified_table(relation_name: &str) -> (String, String) {
+        info!("Validating relation: {}", relation_name);
+
+        let parts = relation_name
+            .split('.')
+            .map(|name| name.to_string())
+            .collect::<Vec<String>>();
+
+        let (schema_name, table_name) = match parts.len() {
+            1 => (None, parts[0].clone()),
+            2 => (Some(parts[0].clone()), parts[1].clone()),
+            _ => error!(
+                "Relation name \"{}\" is not parsable into schema name and table name",
+                relation_name
+            ),
+        };
+
+        match schema_name {
+            None => {
+                let table_count = Spi::get_one_with_args::<i64>("SELECT COUNT(*) FROM information_schema.tables WHERE table_name = $1 AND table_schema = 'public'", vec![
+                    (PgBuiltInOids::TEXTOID.oid(), table_name.clone().into_datum())
+                ]).unwrap();
+
+                let error = format!("Relation \"{}\" could not be found in the public schema. Please specify the table schema, e.g. pgml.{}", table_name, table_name);
```

Comment:
Why doesn't the search path take care of this?

Reply:
`fn analysis`